### ISSUE TITLE
SNOW-641769: Failure to connect when user's default role is lower-case

### ISSUE THREAD

**Issue by jberkenbilt** [snowflakedb/gosnowflake#640]
### Issue description
If a user has a default role that is lower case, you can't connect using gosnowflake.

Note: I have not been able to reproduce this bug from scratch because, when you create a user today, snowflake appears to convert the role to upper case automatically. But my snowflake account has some old users in it whose roles were not upper case.

### Example code

Just pass a fully valid connect string in as the single argument.

```go
package main

import (
	"database/sql"
	"fmt"
	"log"
	"os"

	_ "github.com/snowflakedb/gosnowflake"
)

func main() {
	connectString := os.Args[1]
	sf, err := sql.Open("snowflake", connectString)
	if err != nil {
		log.Fatal(err)
	}

	rows, err := sf.Query("SELECT 'hello'")
	if err != nil {
		log.Fatal(err)
	}
	for rows.Next() {
		var id string
		if err := rows.Scan(&id); err != nil {
			log.Fatal(fmt.Sprintf("Scan error: %v", err))
		}
		fmt.Println(id)
	}
}
```

### Error log
```
ERRO[0001]log.go:242 gosnowflake.(*defaultLogger).Errorln Authentication FAILED                        
2022/08/09 17:07:02 390201 (08004): The requested database does not exist or not authorized.
```

### Configuration
```
module bug

go 1.19

require github.com/snowflakedb/gosnowflake v1.6.12

require (
	github.com/Azure/azure-pipeline-go v0.2.3 // indirect
	github.com/Azure/azure-storage-blob-go v0.14.0 // indirect
	github.com/apache/arrow/go/arrow v0.0.0-20211112161151-bc219186db40 // indirect
	github.com/aws/aws-sdk-go-v2 v1.11.0 // indirect
	github.com/aws/aws-sdk-go-v2/aws/protocol/eventstream v1.0.0 // indirect
	github.com/aws/aws-sdk-go-v2/credentials v1.6.1 // indirect
	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.7.1 // indirect
	github.com/aws/aws-sdk-go-v2/internal/configsources v1.1.0 // indirect
	github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.0.0 // indirect
	github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v1.5.0 // indirect
	github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.5.0 // indirect
	github.com/aws/aws-sdk-go-v2/service/internal/s3shared v1.9.0 // indirect
	github.com/aws/aws-sdk-go-v2/service/s3 v1.19.0 // indirect
	github.com/aws/smithy-go v1.9.0 // indirect
	github.com/form3tech-oss/jwt-go v3.2.5+incompatible // indirect
	github.com/gabriel-vasile/mimetype v1.4.0 // indirect
	github.com/google/flatbuffers v2.0.0+incompatible // indirect
	github.com/google/uuid v1.3.0 // indirect
	github.com/jmespath/go-jmespath v0.4.0 // indirect
	github.com/klauspost/compress v1.13.6 // indirect
	github.com/mattn/go-ieproxy v0.0.1 // indirect
	github.com/pierrec/lz4/v4 v4.1.11 // indirect
	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8 // indirect
	github.com/sirupsen/logrus v1.8.1 // indirect
	golang.org/x/crypto v0.0.0-20211117183948-ae814b36b871 // indirect
	golang.org/x/net v0.0.0-20211118161319-6a13c67c3ce4 // indirect
	golang.org/x/sys v0.0.0-20211117180635-dee7805ff2e1 // indirect
	golang.org/x/text v0.3.7 // indirect
	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
)
```

*Go version:* run `go version` in your console
```
go version go1.19 linux/amd64
```

*Server version:* E.g. 1.90.1
```
6.26.1
```

*Client OS:* Ubuntu 22.04, amd64

Basically if I did a `show user xyz`, I saw the default role as being lower-case. If I just altered the user to change the role to itself, it ends up being upper case. Then it works. Neither snowsql CLI nor the Python drivers care about this.

**Comment by sfc-gh-dszmolka** [comment]
hi, thank you for submitting this issue ! i understand it might be confusing that some python-based clients (SnowSQL , Python connector) do not care about the case of the default_role, but actually this seems to be (or, have been..) an issue with the Snowflake engine. 

The correct behaviour is to prevent the login, and the GUI does the same if you somehow managed to create a lowercase role name and assign it as default_role. The Snowflake-engine internal representation is uppercase so it would be expected for the login to fail.

If you still have users which somehow have been assigned with a lowercased default_role in the past, I suggest rectifying this issue as you already did; re-assigning them with the role which makes it uppercase as it should.
